relay-server/src/actors/processor.rs

```diff
@@ -1459,21 +1389,16 @@ impl EnvelopeProcessorService {
             .sum();
 
         // For limiting the amount of redis calls we make, in case we already passed the limit.
-        if cached_rate_limits
-            .check_with_quotas(quotas, item_scoping)
-            .is_limited()
-        {
-            relay_log::info!("dropping {total_buckets} buckets due to throughput ratelimit");
-            let reason_code = cached_rate_limits
-                .longest()
-                .and_then(|limit| limit.reason_code.clone());
-
-            self.drop_buckets_with_outcomes(
-                reason_code,
-                total_buckets,
+        let limits = cached_rate_limits.check_with_quotas(quotas, item_scoping);
+        if limits.is_limited() {
+            relay_log::debug!("dropping {total_buckets} buckets due to throughput ratelimit");
+
+            let reason_code = limits.longest().and_then(|limit| limit.reason_code.clone());
```

Comment:
The bugfix is here. This uses the returned `limits` from `check_with_quotas` instead of the original `cached_rate_limits`.